crates/ruff_python_formatter/src/format/expr.rs

```diff
@@ -97,62 +96,68 @@ fn format_tuple(
     } else if !elts.is_empty() {
         write!(
             f,
-            [group(&format_with(|f| {
-                if expr.parentheses.is_if_expanded() {
-                    write!(f, [if_group_breaks(&text("("))])?;
-                }
-                if matches!(
-                    expr.parentheses,
-                    Parenthesize::IfExpanded | Parenthesize::Always
-                ) {
-                    write!(
-                        f,
-                        [soft_block_indent(&format_with(|f| {
-                            let magic_trailing_comma =
-                                expr.trivia.iter().any(|c| c.kind.is_magic_trailing_comma());
-                            let is_unbroken =
-                                expr.location.row() == expr.end_location.unwrap().row();
-                            if magic_trailing_comma {
-                                write!(f, [expand_parent()])?;
-                            }
-                            for (i, elt) in elts.iter().enumerate() {
-                                write!(f, [elt.format()])?;
-                                if i < elts.len() - 1 {
-                                    write!(f, [text(",")])?;
-                                    write!(f, [soft_line_break_or_space()])?;
-                                } else {
-                                    if magic_trailing_comma || is_unbroken {
-                                        write!(f, [if_group_breaks(&text(","))])?;
+            [group(&format_with(
+                |f: &mut Formatter<ASTFormatContext<'_>>| {
```

Comment:
CC: @charliermarsh it could be useful to define a `type PyFrmatter<'a> = Formatter<ASTFormatContext<'a>>` type alias.